index.bs

```diff
@@ -575,79 +575,225 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+
+				1. Let |types| be the result of running [=create a frozen array=] of {{DOMString}}.
+
+				1. Let |mimeTypeString| be the result of [=serialize a mime type=] with |mimeType|.
+				
+				1. Add |mimeTypeString| to |types|.
+
+				1. Set |types| to [=this=]'s [=ClipboardItem/types array=].
+
+		<h4 attribute for=ClipboardItem lt=presentationStyle>presentationStyle</h4>
+		<p>
+			The {{ClipboardItem/presentationStyle}} getter steps are to return [=this=]'s [=ClipboardItem/clipboard item=]'s [=presentation style=].
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=types>types</h4>
+		<p>
+			{{ClipboardItem/types}} getter steps are to return [=this=]'s [=ClipboardItem/types array=].
+		</p>
+
+		<h4 method for=ClipboardItem lt=getType(type)>getType(type)</a> must run the below steps:</h4>
+
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |mimeType| be the result of [=parse a mime type=] given |type|.
+
+			1. If |mimeType| is failure, then throw a {{TypeError}}.
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+			1. Let |itemTypeList| be [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+			1. For each |representation| in |itemTypeList|:
+
+				1. If |representation|'s [=mime type=] is |mimeType|, then:
+
+					1. Let |blobData| be a [=Blob=] created using |representation|'s [=data=].
+
+					1. Let |p| be [=a new promise=] in |realm|.
+
+					1. Resolve |p| with |blobData|.
+
+					1. Return |p|.
+
+			1. Return [=a promise rejected with=] {{"NotFoundError"}} DOMException in |realm|.
+
+	<h3 id="clipboard-interface">Clipboard Interface</h3>
+
+		<pre class="idl" data-highlight="webidl">
+		typedef sequence&lt;ClipboardItem&gt; ClipboardItems;
+
+		[SecureContext, Exposed=Window]
+		interface Clipboard : EventTarget {
+			Promise&lt;ClipboardItems&gt; read();
+			Promise&lt;DOMString&gt; readText();
+			Promise&lt;undefined&gt; write(ClipboardItems data);
+			Promise&lt;undefined&gt; writeText(DOMString data);
 		};
 		</pre>
 
-		<div id="clipboard-idl" dfn-for="Clipboard">
+		A <dfn>Clipboard</dfn> may contain <dfn>Clipboard Items</dfn> which has one or more [=clipboard item=]s. Multiple [=clipboard item=]s are only supported on platforms such as iOS/iPadOS.
+		A web author needs to create a |data| which is a [=Clipboard items=] object in order to write content to [=system clipboard=] using the {{Clipboard/write(data)}} method.
+		{{Clipboard/read()}} returns a [=Promise=] to [=Clipboard items=] object that represents contents of [=system clipboard=].
+		A [=clipboard item=] can be read from [=clipboard items=].
 
+		The <dfn>clipboard task source</dfn> is triggered in response to reading or writing of [=system clipboard data=].
+
+		The <dfn>permission task source</dfn> is triggered in response to [=request permission to use=].
```

Comment:
I think you should cross-link to https://w3c.github.io/permissions/#dfn-permissions-task-source instead of defining a new task source. So just deleting this and using `[=permissions task source=]` should work.

Reply:
Ack.